### DISCUSSION TITLE
Need Examples for presolve

### DISCUSSION THREAD

**Discussion by mohit-mhjn** [coin-or/CoinUtils#184] [Q&A] [answered]
I am looking for examples of "how to utilize the presolve methods" 

I have read the file using the `CoinMpsIO` and now I have a `CoinPackedMatrix` and `ObjectiveCoeffs` and `RightHandSide`  I want to load these to the presolve methods and obtain the resultant matrix and vectors. Can someone point me in the right direction? 

I checked cbc but it seems like it doesn't use CoinUtils in a direct way.

**Comment by LouHafer** [comment]
Mohit, doing some work that led to a drive-by touch on CoinUtils presolve, and saw this question. This likely comes long after it's useful to you, but might help someone else in the future. Look at the code in OsiPresolve.cpp. It's pulling data from an OsiSolverInterface, but you can see how to load up a CoinPresolveMatrix, run the presolve transforms, invoke a solver, load up a CoinPostsolveMatrix, and run the postsolve transforms. Also have a look at the doxygen documentation --- classes CoinPresolveMatrix and CoinPresolveAction are good places to start.

**Reply by mohit-mhjn** [reply]
Thanks @LouHafer. I looked at your previous suggestion for OSIPresolve example. It helped to get a sense of how to stich together the components of CoinUtils. I shall do another review through your latest suggestion. Thank you.

**Comment by LouHafer** [comment, marked as answer]
@mohit-mhjn, this likely comes long after you've found another way. I've just come back to playing with COIN-OR code and had to cure a bit of software rot in dylp before I could write this. The previous comment points you to the OsiPresolve code. If you're looking for another example, have a look at OsiDylp, specifically OsiDylpPresolve.cpp. It's another complete example of using the CoinUtils presolve code.
